index.bs

```diff
@@ -5922,6 +5932,144 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
 :: [=largeblob|This extension=] directs the user-agent to cause the large blob to be stored on, or retrieved from, the authenticator. It thus does not specify any direct authenticator interaction for [=[RPS]=].
 
 
+## Device-bound public key extension (<dfn>devicePubKey</dfn>) ## {#sctn-device-publickey-extension}
```

Comment:
> Is it fair to say this extension is only useful in cases where the credential public key registration is performed with direct attestation, and further that this *not* be a self attestation?
>
> Really, what's the point in declaring a device public key if there is no real verifiable root of trust of the authenticator attestation of the associated credential public key?

The DPK is providing a device continuity signal whose usefulness is arguably independent of what sort of attestation of the user credential may have been supplied at the time the user credential public key was originally registered.  

The devicePubKey extension output in conjunction with create() or get() output demonstrates authenticator's possession of both the (potentially synced) user credential private key and the device private key, simultaneously, by employing a nested set of signatures beginning with the authenticator's attestation private key.

> In the case of "none" attestation, there is no AAGUID anyway, so you can't even provide a meaningful value for aaguid in devicePubKey extension output. I would argue that for "self" attestation it's meaningless as well...

"none" is an *[attestation conveyance](https://w3c.github.io/webauthn/#attestation-conveyance)" preference, not an attestation type, and signals the client platform to modify the attestation returned by the authenticator before returning the attstn object to the RP. Thus the authnr may indeed have an AAGUID to employ with the device public key extension.

"self attestation" is an "[attestation type](https://w3c.github.io/webauthn/#sctn-attestation-types)" and (offhand) seems not applicable/appropriate for use in attesting to device public keys.  "Basic" attestation type (aka "batch attestation") is the primarily applicable attestation type.  The AttCA and AnonCA types require further thinking.

Reply:
As an RP, I can certainly receive a ["none" attestation statement format](https://www.w3.org/TR/webauthn-2/#sctn-none-attestation), and that can and does happen at times regardless of the RP-requested attestation conveyance preference. One such example is the current use of Apple passkey credentials. I can't see any point in accepting values of this extension containing a "none" attestation statement format. I think when the TODO part of this PR is being done (`how to verify a attObjForDevicePublicKey`) that we should mention specifically what to accept/reject for allowed attestation statement formats.

Reply:
> As an RP, I can certainly receive a "none" attestation statement format, and that can and does happen at times regardless of the RP-requested attestation conveyance preference. One such example is the current use of Apple passkey credentials. 

Yes, though, the attestation you're referring to there is that of the user credential. As @agl notes in his [above comment](https://github.com/w3c/webauthn/pull/1663#discussion_r743994007): "So the two attestations (user cred's, and DPK's) could happen to be equal, but that wouldn't generally be true.", and so the DPK attestation in this case could be tied to hardware roots of trust even though the user credential's attestation is not.

It will also depend on what DPK attestation types/formats/conveyances we determine are appropriate for DPKs spec-wise (i.e., on a technical basis), and then what a given DPK-supporting authenticator actually implements.

> I can't see any point in accepting values of this extension containing a "none" attestation statement format. I think when the TODO part of this PR is being done (`how to verify a attObjForDevicePublicKey`) that we should mention specifically what to accept/reject for allowed attestation statement formats.

Is it not up to RPs to decide what to trust? E.g., presently in Step 20 of section [7.1. Registering a New Credential](https://w3c.github.io/webauthn/#sctn-registering-a-new-credential) assessing "trustworthiness" is done by the RP according to the RP's policies, and the webauthn spec does not attempt to stipulate such policies. 

Nominally, if an RP receives a "none" DPK attestation from an authenticator (which I suspect could potentially happen regardless of what the spec ends up saying), then it seems it is up to the RP to decide what to do, which could vary, e.g., based on how sophisticated (or not) the RP's session risk scoring is.

Reply:
> Nominally, if an RP receives a "none" DPK attestation from an authenticator (which I suspect could potentially happen regardless of what the spec ends up saying), then it seems it is up to the RP to decide what to do, which could vary, e.g., based on how sophisticated (or not) the RP's session risk scoring is.

Yes it is ultimately up to the RP to decide what to do, however practical guidance can and should be provided as to the verification rules for an RP. I do not find the use of the DPK extension intuitive, and the stronger the guardrails here the better. 

For example I cannot think of any practical scenario in which an RP, having requested the DPK extension, should then accept a DPK attestation that cannot be verified back to a hardware root of trust. After all, that's the entire point of the extension in the first place - provide a device-bound continuity check. A DPK attestation containing attestation statement format of "none" or "self" provides no such assurance.

Reply:
Hardware may not be available all the time on every device user has. Which means that there may not be hardware root of trust all the time. Then it is up to the RP to decide what to do in its risk analysis. What DPK is telling that it is a new device in that case, which is still a requirement to figure out even when there may not be a hardware root of trust.

Reply:
I can logically "infer" or fallback to the posture of it benig a new device if no DPK is present. For example, how does DPK provides any better or more reliable a trust signal over a persistent cookie if it isn't attested?